sklearn/ensemble/weight_boosting.py

```diff
@@ -1047,9 +1050,8 @@ def _boost(self, iboost, X, y, sample_weight):
         estimator_weight = self.learning_rate * np.log(1. / beta)
 
         if not iboost == self.n_estimators - 1:
-            sample_weight *= np.power(
-                beta,
-                (1. - error_vect) * self.learning_rate)
+            sample_weight *= as_float_array(np.power(
+                beta, (1. - error_vect) * self.learning_rate), copy=False)
```

Comment:
Not sure which fix is more preferrable but I had a slightly different fix for the AdaboostRegressor failure, see [this](https://github.com/lesteve/scikit-learn/commit/0f4fde7a490aaee3cc630a0b3e7b560fa60b2149)